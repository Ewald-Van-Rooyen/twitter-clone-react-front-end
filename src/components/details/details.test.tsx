import TestRenderer, {ReactTestRenderer} from "react-test-renderer";
import {render, screen} from "@testing-library/react";
import "@testing-library/jest-dom";
import React from "react";
import Details from "./details";

describe("Tests for the Details component", () => {

    const userDetails = {
        birthday: "1989-08-04",
        firstName: "Ewald",
        lastName: "Van Rooyen",
        tweetsPerDay: 5,
        username: "@Ewald42"
    };

    it("should render the Details screen", () => {
        let testRenderer: ReactTestRenderer = TestRenderer.create(
            <Details birthday={userDetails.birthday}
                     fullName={userDetails.firstName + userDetails.lastName}
                     tweetsPerDay={userDetails.tweetsPerDay}
                     username={userDetails.username}/>
        );
        expect(testRenderer).toMatchSnapshot();
    });

    it("should check the Details screen content", () => {
        render(<Details birthday={userDetails.birthday}
                        fullName={`${userDetails.firstName}  ${userDetails.lastName}`}
                        tweetsPerDay={userDetails.tweetsPerDay}
                        username={userDetails.username}/>);

        expect(screen.getByText(/@Ewald42/i)).toBeInTheDocument();
        expect(screen.getByText(/Ewald Van Rooyen/i)).toBeInTheDocument();
        expect(screen.getByText(/1989-08-04/i)).toBeInTheDocument();
        expect(screen.getByText(/5/i)).toBeInTheDocument();
    });


});
