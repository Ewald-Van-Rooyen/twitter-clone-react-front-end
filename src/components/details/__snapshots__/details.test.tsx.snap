// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests for the Details component should render the Details screen 1`] = `
<div
  className="user-container"
>
  <div
    className="display-flex info-line-item"
  >
    <label
      className="label-main"
    >
      Username:
    </label>
    <label
      className="label-info twitter-color"
    >
      @Ewald42
    </label>
  </div>
  <div
    className="display-flex info-line-item"
  >
    <label
      className="label-main"
    >
      Fullname:
    </label>
    <label
      className="label-info"
    >
      EwaldVan Rooyen
    </label>
  </div>
  <div
    className="display-flex info-line-item"
  >
    <label
      className="label-main"
    >
      Birthday:
    </label>
    <label
      className="label-info"
    >
      1989-08-04
    </label>
  </div>
  <div
    className="display-flex info-line-item"
  >
    <label
      className="label-main"
    >
      Tweets per day over the last 10 days:
    </label>
    <label
      className="label-info"
    >
      5
    </label>
  </div>
</div>
`;
